maths/series/geometric_series.py

```diff
@@ -11,53 +10,54 @@
 """
 
 
-def geometric_series(nth_term: int, start_term_a: int, common_ratio_r: int) -> list:
-    """Pure Python implementation of Geometric Series algorithm
+def geometric_series(
+    nth_term: float, start_term_a: float, common_ratio_r: float
+) -> list[float]:
+    """
+    Pure Python implementation of Geometric Series algorithm
+
     :param nth_term: The last term (nth term of Geometric Series)
     :param start_term_a : The first term of Geometric Series
     :param common_ratio_r : The common ratio between all the terms
     :return: The Geometric Series starting from first term a and multiple of common
         ration with first term with increase in power till last term (nth term)
+
     Examples:
-    >>> geometric_series(4, 2, 2)
-    [2, '4.0', '8.0', '16.0']
     >>> geometric_series(4.0, 2.0, 2.0)
-    [2.0, '4.0', '8.0', '16.0']
+    [2.0, 4.0, 8.0, 16.0]
+    >>> geometric_series(4.0, 2.0, 2.0)
+    [2.0, 4.0, 8.0, 16.0]
     >>> geometric_series(4.1, 2.1, 2.1)
-    [2.1, '4.41', '9.261000000000001', '19.448100000000004']
-    >>> geometric_series(4, 2, -2)
-    [2, '-4.0', '8.0', '-16.0']
-    >>> geometric_series(4, -2, 2)
-    [-2, '-4.0', '-8.0', '-16.0']
-    >>> geometric_series(-4, 2, 2)
```

Comment:
Same.  Please revert these changes.

Reply:
Ok, so make the test floats ints again?

Reply:
The idea is to MIX so 4 as an int and 4.41 as a float and make sure the algorithm does the right thing.